### DISCUSSION TITLE
Compute shader debugging: parent device is lost

### DISCUSSION THREAD

**Discussion by javirk** [gfx-rs/wgpu#3344] [Q&A]
Hi, I have a compute shader that constantly gives me the error:

`thread 'main' panicked at 'Error in Surface::get_current_texture_view: parent device is lost'`

I suppose that this is due to an operation trying to write in the wrong memory address (because that would be the error in that case, right?). However, I really can't find the issue in my code after many days. How could I go into debugging an error like this? I'm trying with RenderDoc, but the error seems random.

[Here](https://github.com/javirk/cell_simulation/blob/master/res/shader/rdme.wgsl) is the shader that is giving the issue, but it's long and probably hard to read for someone not familiar with the whole program. Also, I'm still learning WGSL and GPU programming, there will be noobie mistakes everywhere :D

**Comment by gents83** [comment]
Usually when It happens is because the code Is stuck in an infinite/very long loop or exceeding boundaries while iterating

**Comment by jsonmona** [comment]
You might be hitting the timeout.

If you're on Windows, default timeout is 5 seconds. Any task longer than that will be canceled. This is at driver-level and not something that can be adjusted by wgpu.

You should split your shader, or, set `TdrLevel` registry to 0 if you want to disable the timeout. [relevant msdn docs](https://learn.microsoft.com/en-us/windows-hardware/drivers/display/tdr-registry-keys)
